semantic_conventions/trace/general.yaml

```
@@ -33,6 +33,14 @@ groups:
               brief: 'Something else (non IP-based).'
         brief: >
           Transport protocol used. See note below.
+      - id: app.protocol.name
+        type: string
+        brief: 'Application layer protocol used. The value SHOULD be normalized to lowercase.'
+        examples: ['amqp', 'http', 'mqtt']
+      - id: app.protocol.version
+        type: string
+        brief: 'Version of the application layer protocol used.'
```

```suggestion
        brief: 'Version of the application layer protocol used (might be different from the protocol client's version).'
```

Not sure if that adds clarity or confusion though, your call.

---

It took me some time to understand the motivation behind it, but if other reviewers think it adds clarity, I'm fine adding it.

---

I am not sure what you meant, is "protocol client" the package that is used as the client for this operation?
For example, [axios in npm](https://www.npmjs.com/package/axios/v/0.27.2) is currently version 0.27.2 but the http it produces is 1.1, so this value should be `1.1` and not `0.27.2`?

---

@pyohannes To make it easier to understand you could also make it as `note` where you can add a more detailed explanation and maybe an example like @blumamir brought up.